communities/src/lib.rs

```diff
@@ -92,15 +113,31 @@ decl_module! {
                 validate_nominal_income(&i).map_err(|_| <Error<T>>::InvalidNominalIncome)?;
             }
 
-            let cid = CommunityIdentifier::from(blake2_256(&(loc.clone(), bootstrappers.clone()).encode()));
+            let cid = CommunityIdentifier::from(blake2_256(&(location.clone(), bootstrappers.clone()).encode()));
```

Comment:
I wonder if it makes sense to change the derivation for cid.
Would it make sense to let the first few bytes of the cid type by the geohash? This way we could use the cid value in various ways as it contains information and the bucket could be derived directly. The GeoHash type could directly offer a `fn get_bucket()` and we could drop the storage map `CommunityIdentifiersByGeohash`. less state bloat.
thoughts? (also @clangenb )

I think the geohash should be more precise than `GEO_HASH_LENGTH` used for buckets (AFAIU) because that would make it even easier to diplay pins on a map in encointer/explorer without looking up locations at all

I would still postfix a hash of (location, bootstrappers) to bind these informations

Reply:
nice thing: from the cid alone in ss58 representation you could learn to guess the geographical region of the community

Reply:
and the `Locations`storage could probably become a `map` instead of `double_map`, right?

Reply:
as this would be a fundamental refactoring, I believe we need to decide this first....as the rest of the review will depend on it

Reply:
@brenzi  i am not sure if this is feasible, because in order to find all locations in one bucket, we would have to loop over all cids and do a prefix matching, which is pretty inefficient.

about the geo hash precision: it might be better for visual purposes to increase the geohash precision, but with increasing precision, we would have to do significantly more storage lookups in order to find buckets in a certain radius, which would also make things less efficient again.

Reply:
@brenzi and there is also another flaw in this approach: one community will most of the time spread over more than one geohash bucket

Reply:
I agree so far with @pifragile. I do not see a better alternative to the current storage design. However, I like the direct incorporation of the GeoHash in the cid, which we could do even if we can't use this information.

Reply:
We could argue about the hash precision, but I believe this should be solved client-side.

Reply:
@pifragile I think this is a misunderstanding. I first and foremost just propose to change the cid type. Nothing changes with buckets. Before, it was a meaningless hash, now I suggest to let it start with a geohash. In the minimum this gives us advantages on the UI performance side. This change should not affect anything in your logic. But it might have advantages that I'm not yet clear about, it's more a gut feeling.
I see that `get_bucket` will only lead a single bucket result, not sure if that will work

Reply:
@brenzi, oh i see. i thought you wanted to change the datastructures as well!
is this something you want to consider implementing as part of this PR, or should we leave this as food for thought?

Reply:
Opened issue https://github.com/encointer/pallets/issues/54 for this.